service/controller/app/app.go

```diff
@@ -116,6 +120,12 @@ func NewApp(config Config) (*App, error) {
 			Name: project.Name() + appControllerSuffix,
 		}
 
+		if !config.UniqueApp {
+			// Only watch app CRs in the current namespace. The label selector
+			// excludes the operator's own app CR which has the unique version.
+			c.Namespace = config.PodNamespace
```

Comment:
This is the main change. For the unique instance we continue to watch all namespaces.